winfsp-sys/build.rs

```diff
@@ -27,59 +35,89 @@ fn system() -> String {
+    format!("--include-directory={}inc", directory)
 }
 
 fn main() {
+    // host needs to be windows
+    if !cfg!(windows) {
+        panic!("Compilation requires windows hosts");
+    }
+
+    // dylib feature links to dylib, otherwise links to static
+    #[cfg(feature = "dylib")]
+        let dir = "bin";
+    #[cfg(not(feature = "dylib"))]
+        let dir = "lib";
 
-    #[cfg(feature = "system")] let link_include = system();
-    #[cfg(not(feature = "system"))] let link_include = local();
+    #[cfg(feature = "dylib")]
+        let kind = "dylib";
+    #[cfg(not(feature = "dylib"))]
+        let kind = "static";
+
+    #[cfg(feature = "system")]
+        let link_include = system(dir);
+    #[cfg(not(feature = "system"))]
+        let link_include = local(dir);
 
     println!("cargo:rerun-if-changed=wrapper.h");
 
     if cfg!(target_os = "windows") && cfg!(target_arch = "x86_64") && cfg!(target_env = "msvc") {
-        println!("cargo:rustc-link-lib=dylib=winfsp-x64");
-        println!("cargo:rustc-link-lib=dylib=delayimp");
-        println!("cargo:rustc-link-arg=/DELAYLOAD:winfsp-x64.dll");
+        println!("cargo:rustc-link-lib={kind}=winfsp-x64");
+        if kind == "dylib" {
+            println!("cargo:rustc-link-lib={kind}=delayimp");
+            println!("cargo:rustc-link-arg=/DELAYLOAD:winfsp-x64.dll");
+        }
     } else if cfg!(target_os = "windows") && cfg!(target_arch = "i686") && cfg!(target_env = "msvc") {
-        println!("cargo:rustc-link-lib=dylib=winfsp-x86");
-        println!("cargo:rustc-link-lib=dylib=delayimp");
-        println!("cargo:rustc-link-arg=/DELAYLOAD:winfsp-x86.dll");
+        println!("cargo:rustc-link-lib={kind}=winfsp-x86");
+        if kind == "dylib" {
+            println!("cargo:rustc-link-lib={kind}=delayimp");
+            println!("cargo:rustc-link-arg=/DELAYLOAD:winfsp-x86.dll");
+        }
     } else {
         panic!("unsupported triple")
     }
 
-    let bindings = bindgen::Builder::default()
-        .header("wrapper.h")
-        .derive_default(true)
-        .blocklist_type("_?P?IMAGE_TLS_DIRECTORY.*")
-        .allowlist_function("Fsp.*")
-        .allowlist_type("FSP.*")
-        .allowlist_type("Fsp.*")
-        .allowlist_var("FSP_.*")
-        .allowlist_var("Fsp.*")
-        .allowlist_var("CTL_CODE")
-        .clang_arg("-DUNICODE")
-        .clang_arg(link_include);
-
-    let bindings = if cfg!(target_os = "windows") && cfg!(target_arch = "x86_64") && cfg!(target_env = "msvc") {
-        bindings.clang_arg("--target=x86_64-pc-windows-msvc")
-    } else if cfg!(target_os = "windows") && cfg!(target_arch = "i686") && cfg!(target_env = "msvc") {
-        bindings.clang_arg("--target=i686-pc-windows-msvc")
-    } else {
-        panic!("unsupported triple")
-    };
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let bindings_path_str = out_dir.join("bindings.rs");
+
+    if !Path::new(&bindings_path_str).exists() {
```

Comment:
Let's us cache the bindings.rs, so we don't keep building it again on successive runs.